cypress/e2e/po/pages/explorer/workloads/workloads.po.ts

```diff
@@ -0,0 +1,182 @@
+  ) {
+    const urlStr = `/c/${ clusterId }/explorer/${ workloadType }/${ namespaceId }/${ workloadId }`;
+
+    if (!queryParams) {
+      return urlStr;
+    }
+
+    const params = new URLSearchParams(queryParams);
+
+    return `${ urlStr }?${ params.toString() }`;
+  }
+
+  static goTo(): Cypress.Chainable<Cypress.AUTWindow> {
+    return super.goTo(this.url);
+  }
+
+  constructor(workloadId: string, clusterId: string, workloadType: string, queryParams?: { [key: string]: string }, namespaceId = 'default') {
+    super(workloadDetailsPageBasePo.createPath(workloadId, clusterId, workloadType, namespaceId, queryParams));
+
+    workloadDetailsPageBasePo.url = workloadDetailsPageBasePo.createPath(
+      workloadId,
+      clusterId,
+      workloadType,
+      namespaceId,
+      queryParams
+    );
+  }
+
+  private workload() {
+    return new WorkloadPo();
+  }
+
+  deleteWithKubectl(name: string, namespace = 'default') {
+    this.workload().deleteWithKubectl(name, namespace);
+  }
+
+  createWithKubectl(blueprintJson: string | Object, wait = 6000) {
+    this.workload().createWithKubectl(blueprintJson, wait);
+  }
+}
+
+export class WorkloadsListPageBasePo extends PagePo {
+  static createPath(clusterId: string, workloadType: string, queryParams?: { [key: string]: string }) {
+    const urlStr = `/c/${ clusterId }/explorer/${ workloadType }`;
+
+    if (!queryParams) {
+      return urlStr;
+    }
+
+    const params = new URLSearchParams(queryParams);
+
+    return `${ urlStr }?${ params.toString() }`;
+  }
+
+  constructor(clusterId = 'local', workloadType: string, queryParams?: { [key: string]: string }) {
+    super(WorkloadsListPageBasePo.createPath(clusterId, workloadType, queryParams));
+  }
+
+  navigateToCreatePage() {
+    const baseResourceList = new BaseResourceList(this.self());
+
+    return baseResourceList.masthead().actions().eq(0).click();
+  }
+
+  resourcesList() {
+    return new BaseResourceList(this.self());
+  }
+
+  sortableTable() {
+    return this.resourcesList().resourceTable().sortableTable();
+  }
+
+  deleteItemWithUI(name: string) {
+    this.sortableTable().rowActionMenuOpen(name, 10).getMenuItem('Delete').click();
+
+    const promptRemove = new PromptRemove();
+
+    promptRemove.remove();
+  }
+
+  createWithKubectl(blueprintJson: string | Object, wait = 6000) {
+    this.workload().createWithKubectl(blueprintJson, wait);
+  }
+
+  listElementWithName(name: string) {
+    return this.sortableTable().rowElementWithName(name);
+  }
+
+  goToDetailsPage(elemName: string) {
+    return this.sortableTable().detailsPageLinkWithName(elemName).click();
+  }
+
+  private workload() {
+    return new WorkloadPo();
+  }
+
+  deleteWithKubectl(name: string, namespace = 'default') {
+    this.workload().deleteWithKubectl(name, namespace);
+  }
+}
+
+export class WorkloadsCreatePageBasePo extends PagePo {
+  static createPath(clusterId: string, workloadType: string, queryParams?: { [key: string]: string }) {
```

Comment:
`createPath()` is defined multiple times, so you may avoid code repetition by creating an util or a base class to extend since we are in the OOP world.